### PR TITLE
Fix braindead CSVDiff

### LINKED ISSUE #24828: CSVDiff not catching diffs
## Bug Description

When changing some code, I expected there to a test failure since the result was supposedly changing. But when running `./run_tests` the test was supposedly passing. Upon closer inspection, the test was using CSVDiff and there was indeed a difference between the output and the gold file. If I revert [this PR](https://github.com/idaholab/moose/pull/24633), the failure shows up.

## Steps to Reproduce

With these changes, there should be an error when doing `./run_tests --re vectorpostprocessors/point_value_sampler.test`:

```
diff --git a/test/tests/vectorpostprocessors/point_value_sampler/gold/point_value_sampler_out_point_sample_0001.csv b/test/tests/vectorpostprocessors/point_value_sampler/gold/point_value_sampler_out_point_sample_0001.csv
index c29875d7a3..1d6a83ccc9 100644
--- a/test/tests/vectorpostprocessors/point_value_sampler/gold/point_value_sampler_out_point_sample_0001.csv
+++ b/test/tests/vectorpostprocessors/point_value_sampler/gold/point_value_sampler_out_point_sample_0001.csv
@@ -1,5 +1,4 @@
 id,u,v,x,y,z
 0,0.099999999999971,0.90000000000002,0.1,0.1,0
 1,0.23,0.77000000000001,0.23,0.4,0
-2,0.77999999999996,0.22,0.78,0.2,0
-
+3,0.77999999999996,0.22,0.78,0.2,0
```

## Impact

Very big impact for CI, since bad changes should show in the form of test failures.

### PR DESCRIPTION
**Pull request by milljm**
It wasn't doing anything except making sure you wern't comparing apples to rocks. Modify our custom operator to actually do work.

Create Unittests, in the hopes it will never go unchecked again.

Added new feature to test absolute differences. This requires relative error/tolerance to be set at zero in order to trigger new behavior, as so:

```conf
[Test]
  type = (All SchemaDiff Types including CSVDIFF)
  rel_err = 0
  abs_err = 5.5e-6

  # Or a specific field (CSV type tests only)
  type = CSVDIFF
  override_columns = z_field  #
  override_rel_err = 0        #
  override_abs_err = 5.5e-6   # for absolute diffing on z_field
[]
```

Closes #24828

### CODE REVIEW

**Comment by loganharbour** [comment]
Did you mean to update large_media?

File: test/tests/test_harness/csvdiff_comparison_floor.cmp [hunk 1]
```diff
@@ -1,7 +1,7 @@
 TIME STEPS relative 5.5e-06 floor 1e-11  # min: 0 @ t0  max: 0 @ t5
 
 GLOBAL VARIABLES relative 5.5e-06 floor 1e-11
-    z_field floor 1e+01        # min: 0.000e+00 @ t0          max: 4.025e-01 @ t5
+    z_field absolute 1e+01     # min: 0.000e+00 @ t0          max: 4.025e-01 @ t5
```
Comment (milljm) [inline test/tests/test_harness/csvdiff_comparison_floor.cmp:4]: hmmm... this doesn't look right.

File: python/TestHarness/testers/CSVDiff.py [hunk 2]
```diff
@@ -7,114 +7,258 @@
...
 
-            if self.specs.isValid('ignore_columns'):
-                csvdiff.append('--ignore-fields %s' % (' '.join(self.specs['ignore_columns'])))
+        # Set nested loop values so variables can be used for error printing
+        value_id = 0
+        value = ''
+        for index, column in enumerate(orig):
+            _rel_err = abs(rel_err)
+            _abs_err = abs(abs_err)
+            _abs_zero = abs(abs_zero)
```
Comment (loganharbour) [inline python/TestHarness/testers/CSVDiff.py:145]: Why underscore prefix here? This is not intuitive to me, as we usually use this prefix for member vars. I would prefer a name that provides more context

**Review by loganharbour — CHANGES_REQUESTED** [review]
Looks great to me. the `do_xxx` methods in the operator make life nice

File: python/TestHarness/testers/SchemaDiff.py [hunk 3]
```diff
@@ -30,7 +34,7 @@ def validParams():
         params.addParam('rel_err',      5.5e-6, 'Relative error value allowed in comparisons. If '
```
Comment (loganharbour) [inline python/TestHarness/testers/SchemaDiff.py:34]: Remove rel_err and abs_zero because they come from FileTester

**Review by loganharbour — CHANGES_REQUESTED** [review]

File: python/TestHarness/testers/SchemaDiff.py [hunk 4]
```diff
@@ -151,8 +154,29 @@ def format_diff(self, diff_dict):
...
+                                         f'\n\tRESULT: {",".join(nodes[1:])}: {computed}\n')
+
         return formated_results
 
+    def compare_func(self, x, y, level):
+        """
+        Our custom iterator. Needed for what we believe is a bug
+        See issue: https://github.com/seperman/deepdiff/issues/404
+        """
+        return True
```
[the remark this answers is not in the document]
Comment (loganharbour) [reply python/TestHarness/testers/SchemaDiff.py:174]: I think this is my only remaining concern. I don't want to change the behavior for JSONDiff/XMLDiff unless we really know it's a problem there too. In those cases, it's a more complex diff so maybe it doesn't suffer the same issues?

[the remark this answers is not in the document]
Comment (milljm) [reply python/TestHarness/testers/SchemaDiff.py:174]: I'll make the change, but consider this:

Wouldn't the same bug exhibit itself in json/xml data? I am passing the same data type that JSON does when performing CSV tests (dictionary of lists).

[the remark this answers is not in the document]
Comment (loganharbour) [reply python/TestHarness/testers/SchemaDiff.py:174]: Maybe. I'm just very hesitant to change everything but Exodiff in this PR

**Pull request closed without merging**